src/util/HashSet.h

```diff
@@ -32,4 +37,88 @@ template <class T,
 using HashSetWithMemoryLimit =
     absl::flat_hash_set<T, HashFct, EqualElem, Alloc>;
 
+template <class T,
+          class HashFct = absl::container_internal::hash_default_hash<T>,
+          class EqualElem = absl::container_internal::hash_default_eq<T>,
+          class SizeGetter = std::function<size_t(const T&)>>
+class CustomHashSetWithMemoryLimit {
+ private:
+  absl::node_hash_set<T, HashFct, EqualElem> hashSet_;
+  detail::AllocationMemoryLeftThreadsafe memoryLeft_;
+  MemorySize memoryUsed_;
+  SizeGetter sizeGetter_;
+
+ public:
+  CustomHashSetWithMemoryLimit(MemorySize memory, SizeGetter sizeGetter)
+      : memoryLeft_(makeAllocationMemoryLeftThreadsafeObject(memory)),
+        memoryUsed_(MemorySize::bytes(0)),
+        sizeGetter_(sizeGetter) {}
```

Comment:
The destructor (which has to free all memory to the memory tracker) is missing. (important once you have an external memory limit).